src/transformer/events/core/question_created.php

```diff
@@ -0,0 +1,74 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Transformer fn for question created event.
+ *
+ * @package   logstore_xapi
+ * @copyright Milt Reder <milt@yetanalytics.com>
+ *
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace src\transformer\events\core;
+
+use src\transformer\utils as utils;
+use src\transformer\utils\get_activity\definition\question as qdef;
+
+/**
+ * Transformer fn for question created event.
+ *
+ * @param array $config The transformer config settings.
+ * @param \stdClass $event The event to be transformed.
+ * @return array
+ */
+
+function question_created(array $config, \stdClass $event) {
+    $repo = $config['repo'];
+    $user = $repo->read_record_by_id('user', $event->userid);
+    $course = $repo->read_record_by_id('course', $event->courseid);
+    $question = $repo->read_record_by_id('question', $event->objectid);
+    $lang = utils\get_course_lang($course);
+
+    $definition = qdef\get_definition($config, $question, $lang);
+
+    $definition['extensions']['https://xapi.edlm/profiles/edlm-lms/concepts/activity-extensions/moodle-question-type'] = $question->qtype;
```

Comment:
why not just toss this into definition fns? In this case we're always making definition FOR moodle so its always relevant. I just see it getting forgotten somewhere otherwise, not that its super important.

Reply:
yeah, I probably should, just look at the source for get_definition and you'll see why I was lazy. I'll look at it again.